frontend/src/components/worksheets/items/ImageItem.js

```diff
@@ -1,23 +1,49 @@
 import * as React from 'react';
-import Immutable from 'seamless-immutable';
-import { worksheetItemPropsChanged } from '../../../util/worksheet_utils';
+import { withStyles } from '@material-ui/core/styles';
+import * as Mousetrap from '../../../util/ws_mousetrap_fork';
+import BundleDetail from '../BundleDetail';
 
 class ImageItem extends React.Component {
     /** Constructor. */
     constructor(props) {
         super(props);
-        this.state = Immutable({});
+        this.state = {
+            showDetail: false,
+            bundleInfoUpdates: {},
+        };
     }
 
+    receiveBundleInfoUpdates = (update) => {
+        let { bundleInfoUpdates } = this.state;
+        // Use object spread to update.
+        bundleInfoUpdates = { ...bundleInfoUpdates, ...update };
+        this.setState({ bundleInfoUpdates });
+    };
+
     handleClick = () => {
         this.props.setFocus(this.props.focusIndex, 0);
+        this.setState({ showDetail: !this.state.showDetail });
     };
 
-    shouldComponentUpdate(nextProps, nextState) {
-        return worksheetItemPropsChanged(this.props, nextProps);
-    }
-
     render() {
+        if (this.props.focused) {
+            // Use e.preventDefault to avoid openning selected link
+            Mousetrap.bind(
+                ['enter'],
+                (e) => {
+                    e.preventDefault();
+                    if (!this.props.confirmBundleRowAction(e.code)) {
+                        this.setState({ showDetail: !this.state.showDetail });
+                    }
+                },
+                'keydown',
+            );
+            // unbind shortcuts that are active for markdown_block and worksheet_block
+            Mousetrap.unbind('i');
```

Comment:
Is `i` the only shortcut that needs to be unbound here?

Reply:
Actually it's not needed...copied from the bundle row implementation